engine/access/rest/routes/websocket_handler.go

```diff
@@ -158,6 +159,20 @@ func (wsController *WebsocketController) writeEvents(sub state_stream.Subscripti
 				blocksSinceLastMessage = 0
 			}
 
+			// EventsResponse contains CCF encoded events, and this API returns JSON-CDC events.
+			// convert event payload formats.
+			events := make([]flow.Event, len(resp.Events))
+			for i, e := range resp.Events {
+				payload, err := convert.CcfPayloadToJsonPayload(e.Payload)
+				if err != nil {
+					err = fmt.Errorf("could not convert event payload from CCF to Json: %w", err)
+					wsController.wsErrorHandler(err)
+					return
+				}
+				events[i].Payload = payload
```

Comment:
This is a event that is only updated with the Payload field, and the rest fields are all empty?

Reply:
no, this just modifies the `Payload` field to be JSON encoded, and leaves the rest of the fields as they were

Reply:
good catch

Reply:
they passed because the handler was overwriting the expected results data. fixed in [5eb72b7](https://github.com/onflow/flow-go/pull/5048/commits/5eb72b7ab5251383681c8daf5228b436d9ed65f9)